content/blog/2016/2016-10-11-stage-lock-milestone.adoc

```diff
@@ -74,7 +74,7 @@ lock("myResource"){
 }
 ----
 
-=== milestone
+=== Milestone
 
 The `milestone` step is the last piece of the puzzle to replace functionality originally intended for `stage` and adds even more control for handling concurrent builds of a job. The `lock` step limits the number of builds running concurrently in a section of your Pipeline while the `milestone` step ensures that older builds of a job will not overwrite or clobber a newer build.
 
```

Comment:
It will add numbers based how it encounters them in sequence. This example is kind of a straw man that doesn't actually do anything. It would still work but would probably end up aborting jobs that you didn't mean to abort.

`// returns a value between 1 and 10
def a = getData()
milestone //1
if (a > 5) {
    milestone //2
}
milestone //3`

Build A : a = 7
Build B: a = 3

Build B has a much better chance of reaching milestone 3 first and aborting Build A.

In general it would be a pretty poor practice to put a milestone within a conditional statement because the point is to compare two builds at common milestones. 

I think introducing  esoteric use cases is beyond the scope of a blog but should be mentioned in the documentation.